src/gobprepare/importers/dump_importer.py

```diff
@@ -0,0 +1,107 @@
+import contextlib
+from io import StringIO
+import tempfile
+import gzip
+import re
+import tempfile
+from typing import List, Iterator, Optional
+
+from gobprepare.importers.typing import ReadConfig, SqlDumpImporterConfig
+from gobconfig.datastore.config import get_datastore_config
+from gobcore.datastore.factory import DatastoreFactory
+from gobcore.datastore.objectstore import ObjectDatastore
+from gobcore.datastore.sql import SqlDatastore
+from gobcore.logging.logger import logger
+
+from gobcore.exceptions import GOBException
+
+@contextlib.contextmanager
+def _load_from_objectstore(datastore: ObjectDatastore) -> Iterator[str]:
+    datastore.connect()
+
+    obj_info = next(datastore.query(None), None)
+
+    if obj_info is None:
+        raise GOBException(f"File not found on Objectstore: {datastore.read_config['file_filter']}")
+
+    _, obj = datastore.connection.get_object(
+        container=datastore.container_name, obj=obj_info["name"], resp_chunk_size=100_000_000
+    )
+
+    with tempfile.NamedTemporaryFile(mode="wb", suffix=".gz") as fp:
+        try:
+            for chunk in obj:
+                fp.write(chunk)
+            logger.info(f"File loaded from objectstore: {obj_info['name']}")
+            yield fp.name
+        finally:
+            datastore.disconnect()
+
+
+class SqlDumpImporter:
+    """Import a SQL dump into a SqlDatastore table."""
+
+    def __init__(self, dst_datastore: SqlDatastore, config: SqlDumpImporterConfig) -> None:
+        """ Initialise SqlDumpImporter """
+        self._dst_datastore = dst_datastore
+        # self._destination = config["destination"]
+        self._encoding = config.get("encoding", "utf-8")
+
+        self._read_config: Optional[ReadConfig] = config.get("read_config")
+        self._objectstore: Optional[str] = config.get("objectstore")
+        self._source: Optional[str] = config.get("source")
+
+    def _process_queries(self, queries: list) -> None:
+        for query in queries:
+            # remove leading new line '\n\n'
+            query = query.lstrip()
+            # filter out and replace the queries based on the filter and substitution lists
+            if (query and not re.match(self._read_config['filter_list'], query)):
+                if not query.endswith(self._read_config['data_delimiter_regexp']):
+                    query += ';'
+                    for pattern, replacement in self._read_config['substitution'].items():
+                        query = re.sub(pattern, replacement, query)
+
+                if re.match(self._read_config['copy_query_regex'], query):
```

Comment:
this regex doesn't change, compiling first increases performance

Reply:
Done.